sdk/identity/identity/src/credentials/managedIdentityCredential/tokenExchangeMsi.ts

```diff
@@ -13,38 +13,36 @@ const logger = credentialLogger(msiName);
 /**
  * Defines how to determine whether the token exchange MSI is available, and also how to retrieve a token from the token exchange MSI.
  */
-export function tokenExchangeMsi(): MSI {
-  return {
-    name: "tokenExchangeMsi",
-    async isAvailable({ clientId }): Promise<boolean> {
-      const env = process.env;
-      const result = Boolean(
-        (clientId || env.AZURE_CLIENT_ID) &&
-          env.AZURE_TENANT_ID &&
-          process.env.AZURE_FEDERATED_TOKEN_FILE,
+export const tokenExchangeMsi: MSI = {
+  name: "tokenExchangeMsi",
+  async isAvailable({ clientId }): Promise<boolean> {
+    const env = process.env;
+    const result = Boolean(
+      (clientId || env.AZURE_CLIENT_ID) &&
+        env.AZURE_TENANT_ID &&
+        process.env.AZURE_FEDERATED_TOKEN_FILE,
+    );
+    if (!result) {
+      logger.info(
+        `${msiName}: Unavailable. The environment variables needed are: AZURE_CLIENT_ID (or the client ID sent through the parameters), AZURE_TENANT_ID and AZURE_FEDERATED_TOKEN_FILE`,
       );
-      if (!result) {
-        logger.info(
-          `${msiName}: Unavailable. The environment variables needed are: AZURE_CLIENT_ID (or the client ID sent through the parameters), AZURE_TENANT_ID and AZURE_FEDERATED_TOKEN_FILE`,
-        );
-      }
-      return result;
-    },
-    async getToken(
-      configuration: MSIConfiguration,
-      getTokenOptions: GetTokenOptions = {},
-    ): Promise<AccessToken | null> {
-      const { scopes, clientId } = configuration;
-      const identityClientTokenCredentialOptions = {};
-      const workloadIdentityCredential = new WorkloadIdentityCredential({
-        clientId,
-        tenantId: process.env.AZURE_TENANT_ID,
-        tokenFilePath: process.env.AZURE_FEDERATED_TOKEN_FILE,
-        ...identityClientTokenCredentialOptions,
-        disableInstanceDiscovery: true,
-      } as WorkloadIdentityCredentialOptions);
-      const token = await workloadIdentityCredential.getToken(scopes, getTokenOptions);
-      return token;
-    },
-  };
-}
+    }
+    return result;
+  },
+  async getToken(
+    configuration: MSIConfiguration,
+    getTokenOptions: GetTokenOptions = {},
+  ): Promise<AccessToken | null> {
+    const { scopes, clientId } = configuration;
+    const identityClientTokenCredentialOptions = {};
+    const workloadIdentityCredential = new WorkloadIdentityCredential({
+      clientId,
+      tenantId: process.env.AZURE_TENANT_ID,
+      tokenFilePath: process.env.AZURE_FEDERATED_TOKEN_FILE,
+      ...identityClientTokenCredentialOptions,
+      disableInstanceDiscovery: true,
+    } as WorkloadIdentityCredentialOptions);
+    const token = await workloadIdentityCredential.getToken(scopes, getTokenOptions);
```

Comment:
I assume the await here is so we can get tracing time for how long getToken took? Still feels a little weird since it returns a promise.

Reply:
hmmm good catch! Honestly I just blindly turned this from a function (`tokenExchangeMsi()`) to a const (`tokenExchangeMsi`) so it matches the rest of the MSIs but without changing the implementation.

Will definitely update this to just return the promise, I don't think there's a reason to await it 👍